algorithms/linfa-bayes/src/error.rs

```diff
@@ -5,7 +5,7 @@ use thiserror::Error;
 pub type Result<T> = std::result::Result<T, NaiveBayesError>;
 
 /// Error variants from hyper-parameter construction or model estimation
-#[derive(Error, Debug)]
+#[derive(Error, Debug, Clone)]
```

Comment:
Extra traits on error types wasn't part of the original issue. Those traits are generally not used for errors and make them harder to extend in the future. I'd like the extra derives on all error types to be reverted.

Reply:
Yes, it's true. I thought it can be useful to add, especially `Clone` (it is derived on `linfa::Error` and some other errors). But removed all extra traits